src/main/java/io/anserini/kg/freebase/EntityLinking.java

```diff
@@ -0,0 +1,360 @@
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.similarities.ClassicSimilarity;
+import org.apache.lucene.search.similarities.Similarity;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * Lookups a Freebase mid and returns all properties associated with it.
+ */
+public class EntityLinking implements Closeable {
+  private static final Logger LOG = LogManager.getLogger(EntityLinking.class);
+
+  private final IndexReader reader;
+
+  private static int numHits = 200;
+
+  static final class Args {
+    // Required arguments
+
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    public String index;
+
+    @Option(name = "-data", metaVar = "[file]", required =  true, usage = "dataset file")
+    public String data;
+
+    @Option(name = "-goldData", usage = "supplying with the ground truth dataset")
+    boolean goldData;
+
+    @Option(name = "-hits", metaVar = "hits", usage = "setting number of hits to be retrieved per query")
+    int hits;
+  }
+
+  class RankedEntity implements Comparable<RankedEntity> {
+    String mid;
+    float score;
+    String name;
+    String label;
+
+    RankedEntity(String m, float s, String n, String l) {
+      mid = m;
+      score = s;
+      name = n;
+      label = l;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("(%s, %f)", mid, score);
+    }
+
+    @Override
+    public int compareTo(RankedEntity o) {
+      return new Float(this.score).compareTo(new Float(o.score));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      boolean result = false;
+      if (other instanceof RankedEntity) {
+        RankedEntity that = (RankedEntity) other;
+        result = that.mid.equals(this.mid);
+      }
+      return result;
+    }
+  }
+
+  private EntityLinking(String indexDir) throws IOException {
+    // Initialize index reader
+    LOG.info("Reading index from " + indexDir);
+
+    Path indexPath = Paths.get(indexDir);
+
+    if (!Files.exists(indexPath) || !Files.isDirectory(indexPath) || !Files.isReadable(indexPath)) {
+      throw new IllegalArgumentException(indexDir + " does not exist or is not a directory.");
+    }
+
+    this.reader = DirectoryReader.open(FSDirectory.open(indexPath));
+  }
+
+  @Override
+  public void close() throws IOException {
+    reader.close();
+  }
+
+  public void searchFile(String fileName) throws Exception {
+    LOG.info("fileName: " + fileName);
+
+    // Open the file
+    FileInputStream fstream = new FileInputStream(fileName);
+    BufferedReader br = new BufferedReader(new InputStreamReader(fstream));
+
+    String strLine;
+
+    //Read File Line By Line
+    int found = 0, notfound = 0;
```

Comment:
Process the file in a try-with-resource block so that the resource is closed after it's being processed.